lib/openscap_parser/group.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Group < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title') &&
+        parsed_xml.at_css('title').text
+    end
+
+    def description
+      @description ||= newline_to_whitespace(
+        parsed_xml.at_css('description') &&
+          parsed_xml.at_css('description').text || ''
+      )
+    end
+
+    def rationale
+      @rationale ||= newline_to_whitespace(
+        parsed_xml.at_css('rationale') &&
+          parsed_xml.at_css('rationale').text || ''
+      )
+    end
+
+    def requires
+      @requires ||= parsed_xml.xpath('./requires') &&
+        parsed_xml.xpath('./requires/@idref').to_s&.split
+    end
+
+    def conflicts
+      @conflicts ||= parsed_xml.xpath('./conflicts') &&
+        parsed_xml.xpath('./conflicts/@idref').to_s
+    end
+
+    def selected
+      @selected ||= parsed_xml['selected']
+    end
+
+    def parent_id
+      @parent_id = parsed_xml.xpath("name(..)='Group'") ?
```

Comment:
How is this handling polymorphism? That a parent can be the `Profile` itself as well?

Reply:
True, good point! I was thinking that if a Group or Rule didn't have a parent_id then we would know that it wasn't nested inside another group but it would be better to set the parent id no matter what.
 I think that it's parent would be a `Benchmark` though, right? Instead of a `Profile`?